### ISSUE TITLE
🐛 Bug Report: Failed to create photo preview immediatelly after upload

### ISSUE THREAD

**Issue by step135** [appwrite/appwrite#7450]
### 👟 Reproduction steps

Upload bigger photo 9MB that fires Webhook calling to another website using newly uploaded URL of photo `https://cloud.appwrite.io/v1/storage/buckets/photos/files/{$id}/preview?project={$prId}`. It creates wrong cache:
![preview](https://github.com/appwrite/appwrite/assets/47241203/b232d059-da42-4774-b599-f10449740c1e)

### 👍 Expected behavior

I should work with uploaded photo.

### 👎 Actual Behavior

It return general png image.

### 🎲 Appwrite version

Appwrite Cloud

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by step135** [comment]
It might be caused that Webhook fires before the upload is finished!

**Comment by eldadfux** [comment]
@lohanidamodar could it be that we fired the event for the first chunk, and that caused the issue before the upload completed? Do we fire an event for other chunks or able to give an indication for the last chunk instead of the first if its a multi-chunk upload with bigger files?

**Comment by lohanidamodar** [comment]
@eldadfux yes, since we moved to labels for events, the event I believe fires for every chunk, we can do the checks here based on Payload to check if the chunks uploaded matches the total chunks and only then fire the event
https://github.com/appwrite/appwrite/blob/14e81c2e27c83a15508fd1806ae98f55570f2d39/app/controllers/shared/api.php#L440

**Comment by step135** [comment]
Is there any progress on this bug? Or still not possible to track when the whole file was created (all chunks uploaded)?

**Comment by stnguyen90** [comment]
@step135 this looks like the same as https://github.com/appwrite/appwrite/issues/4507 so I'm going to close it. It would be best if you add a 👍🏼 to the other issue.

To handle the chunks, your webhook handler should look at the payload and check the chunks uploaded compared to the chunks total. If they match, that should mean the full file has been created.
